src/lib/components/TimeSeriesSlider/TimeSeriesSlider.js

```diff
@@ -161,10 +166,17 @@ class TimeSeriesSlider extends React.Component {
   }
 
   handleMouseUp(e) {
+    e.persist();
     const nextIndex = parseInt(e.target.value, 10);
     const { index, currentYear } = this.state;
+    const self = this;
+    self.e = e;
```

Comment:
@KipSigei - Can we namespace `self.e` to describe what event is being persisted? Also, can we grab only the data we need from the event so we don't need to persist it?

Reply:
we actually don't need to bind event to self, I was having issues when trying to pass in event to `updateTimeseriesState` because for some reason in react when the event handler is invoked the event props i.e target end up getting nullified hence the reason I added `e.persist()` for it to be reused in `updateTimeseriesState` so I will refactor this shortly.